### ISSUE TITLE
Recovery screen not available in agent 1.5.12

### ISSUE THREAD

**Issue by steamraven** [UltimateHackingKeyboard/agent#1496]
When using the latest agent available for download, when UHK is soft reset (shorting reset pins) https://ultimatehackingkeyboard.com/repair/unbricking-the-right-keyboard-half

The recovery screen is not shown.

In 1.5.11, this screen is shown for a couple seconds.

**Comment by mondalaci** [comment]
Are you sure it's Agent? I've just looked into this issue, and it seems to me that the bootloader timeout of the right keyboard half has changed from 5 seconds to an unnoticeably short period from firmware 8.10.6 to 8.10.7.

**Comment by steamraven** [comment]
So this is really weird.  The official firmware does not seem to wait in bootloader after soft reset. However built clean from the master branch tagged v8.10.9 does wait the 5 seconds.  The bootloader determines how long it stays active and we don't overwrite the bootloader, right? Maybe a bootloader config option, I'll open an issue on the firmware side and investigate.

However, even when it waits, the v 1.5.12 just displays "Cannot find UHK" rather than "Fix Device".

It appears commit eb94c5d2b91ac2efef5c47352bcbbd43bfe15834 is the culprit. I don't know why yet

**Comment by steamraven** [comment]
I think I figured out the instantaneous bootloader.  When flashing from the agent, after it completes right, left, and modules, then soft resets are instantaneous.  However, a power cycle will restore the keyboard back to a 5 second soft reset.  I am betting this is a wormhole setting that is left set.  Not sure its severe enough to warrent a agent issue, so at least this mystery is solved.

Not to find out why the agent isn't noticing the bootloader for recovery....

**Comment by mondalaci** [comment]
I take back what I said. I retested the newest 8.10.9 firmware and when shorting the reset pin, it waited for 5 seconds as expected.

We know that the bootloader has a wormhole-related bug, which makes it disrespect the provided timeout, but we don't think it's serious usability-wise, and it can't be fixed on existing UHKs without a dedicated USB programmer dongle, so we ended up ignoring it to make the behavior consistent.

I can confirm that Agent 1.5.12 is the first affected Agent version in which the recovery screen doesn't work. We'll fix this.
